src/test/unit/gov/nist/javax/sip/stack/DialogEarlyStateTimeoutWithout100Test.java

```diff
@@ -0,0 +1,479 @@
+                        .createContentTypeHeader("application", "sdp");
+
+                // Create a new CallId header
+                CallIdHeader callIdHeader = provider.getNewCallId();
+                // JvB: Make sure that the implementation matches the
+                // messagefactory
+                callIdHeader = headerFactory.createCallIdHeader(callIdHeader
+                        .getCallId());
+
+                // Create a new Cseq header
+                CSeqHeader cSeqHeader = headerFactory.createCSeqHeader(1L,
+                        Request.INVITE);
+
+                // Create a new MaxForwardsHeader
+                MaxForwardsHeader maxForwards = headerFactory
+                        .createMaxForwardsHeader(70);
+
+                // Create the request.
+                Request request = messageFactory.createRequest(requestURI,
+                        Request.INVITE, callIdHeader, cSeqHeader, fromHeader,
+                        toHeader, viaHeaders, maxForwards);
+                // Create contact headers
+
+                // Create the contact name address.
+                SipURI contactURI = addressFactory.createSipURI(fromName,
+                        myAddress);
+                contactURI.setPort(provider.getListeningPoint("udp").getPort());
+
+                Address contactAddress = addressFactory
+                        .createAddress(contactURI);
+
+                // Add the contact address.
+                contactAddress.setDisplayName(fromName);
+
+                ContactHeader contactHeader = headerFactory
+                        .createContactHeader(contactAddress);
+                request.addHeader(contactHeader);
+
+                String sdpData = "v=0\r\n"
+                        + "o=4855 13760799956958020 13760799956958020"
+                        + " IN IP4  129.6.55.78\r\n"
+                        + "s=mysession session\r\n" + "p=+46 8 52018010\r\n"
+                        + "c=IN IP4  129.6.55.78\r\n" + "t=0 0\r\n"
+                        + "m=audio 6022 RTP/AVP 0 4 18\r\n"
+                        + "a=rtpmap:0 PCMU/8000\r\n"
+                        + "a=rtpmap:4 G723/8000\r\n"
+                        + "a=rtpmap:18 G729A/8000\r\n" + "a=ptime:20\r\n";
+
+                request.setContent(sdpData, contentTypeHeader);
+
+                // The following is the preferred method to route requests
+                // to the peer. Create a route header and set the "lr"
+                // parameter for the router header.
+
+                Address address = addressFactory.createAddress("<sip:"
+                        + PEER_ADDRESS + ":" + PEER_PORT + ">");
+                // SipUri sipUri = (SipUri) address.getURI();
+                // sipUri.setPort(PEER_PORT);
+
+                RouteHeader routeHeader = headerFactory
+                        .createRouteHeader(address);
+                ((SipURI) address.getURI()).setLrParam();
```

Comment:
More of a question than a comment. Why do we need to add the LR parameter here? Isnt this for load balancing purposes?

Reply:
It's by specification, this flag is used to trigger the routing mechanism of the 3261 vs old 2543